plugins/query-resource-limit-group/src/main/java/org/opensearch/plugin/resource_limit_group/UpdateResourceLimitGroupRequest.java

```diff
@@ -0,0 +1,159 @@
+        this.resourceLimits = resourceLimitGroup.getResourceLimits();
+        this.enforcement = resourceLimitGroup.getEnforcement();
+        this.updatedAt = resourceLimitGroup.getUpdatedAt();
+    }
+
+    /**
+     * Constructor for UpdateResourceLimitGroupRequest
+     * @param in - A {@link StreamInput} object
+     */
+    public UpdateResourceLimitGroupRequest(StreamInput in) throws IOException {
+        super(in);
+        name = in.readString();
+        if (in.readBoolean()) {
+            resourceLimits = in.readList(ResourceLimit::new);
+        }
+        enforcement = in.readOptionalString();
+        updatedAt = in.readString();
+    }
+
+    @Override
+    public UpdateResourceLimitGroupRequest read(StreamInput in) throws IOException {
+        return new UpdateResourceLimitGroupRequest(in);
+    }
+
+    /**
+     * Generate a UpdateResourceLimitGroupRequest from XContent
+     * @param parser - A {@link XContentParser} object
+     */
+    public static UpdateResourceLimitGroupRequest fromXContent(XContentParser parser) throws IOException {
+        ResourceLimitGroup resourceLimitGroup = ResourceLimitGroup.fromXContentOptionalFields(parser);
+        return new UpdateResourceLimitGroupRequest(resourceLimitGroup);
+    }
+
+    @Override
+    public ActionRequestValidationException validate() {
+        return null;
+    }
+
+    /**
+     * name getter
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * name setter
+     * @param name - name to be set
+     */
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    /**
+     * ResourceLimits getter
+     */
+    public List<ResourceLimit> getResourceLimits() {
+        return resourceLimits;
+    }
+
+    /**
+     * ResourceLimits setter
+     * @param resourceLimits - ResourceLimit to be set
+     */
+    public void setResourceLimits(List<ResourceLimit> resourceLimits) {
+        this.resourceLimits = resourceLimits;
+    }
+
+    /**
+     * Enforcement getter
+     */
+    public String getEnforcement() {
+        return enforcement;
+    }
+
+    /**
+     * Enforcement setter
+     * @param enforcement - enforcement to be set
+     */
+    public void setEnforcement(String enforcement) {
+        this.enforcement = enforcement;
+    }
+
+    /**
+     * updatedAt getter
+     */
+    public String getUpdatedAt() {
+        return updatedAt;
+    }
+
+    /**
+     * updatedAt setter
+     * @param updatedAt - updatedAt to be set
+     */
+    public void setUpdatedAt(String updatedAt) {
+        this.updatedAt = updatedAt;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeString(name);
+        if (resourceLimits == null || resourceLimits.isEmpty()) {
+            out.writeBoolean(false);
+        } else {
+            out.writeBoolean(true);
+            out.writeList(resourceLimits);
```

Comment:
How about writing `out.writeOptionalArray` api ?

Reply:
Since resourceLimits a list we can't use `out.writeOptionalArray`